File: C++/Recursion/CheckSortedArray.cpp
#include <iostream>
using namespace std;

bool isSorted(int arr[], int i, int n, int choice);

int main()
{
    int n, choice;
    cout << "Enter the length of the Array -> ";
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++)
        cin >> arr[i];
    cout << "In which order do you want to check if Array is sorted or not?\n1 for Ascending\n2 for Descending\n-> ";
    cin >> choice;
    printf("Is Array Sorted? -> %s", isSorted(arr, 0, n, choice) ? "true" : "false");
}

bool isSorted(int arr[], int i, int n, int choice)
{
    if (i == n - 1)
        return true;
    if (choice == 1)
    {
        if (arr[i] < arr[i + 1])
            return isSorted(arr, i + 1, n, choice);
        else
            return false;
    }
    else
    {
        if (arr[i] > arr[i + 1])
            return isSorted(arr, i + 1, n, choice);
        else
            return false;
    }
}

File: Python/recursion/check_sorted_array.py
def is_sorted(arr, i: int, choice: int) -> bool:
    if i == len(arr) - 1:
        return True
    # For Strictly Ascending
    if choice == 1:
        return is_sorted(arr, i + 1, choice) if arr[i] < arr[i + 1] else False
    # For Strictly Descending
    return is_sorted(arr, i + 1, choice) if arr[i] > arr[i + 1] else False


def main():
    arr = []
    n = int(input("Enter the lenght of the Array -> "))
    for _ in range(0, n):
        arr.append(int(input()))
    choice = int(
        input(
            "In which order do you want to check if Array is sorted or not?\n"
            "1 for Ascending\n"
            "2 for Descending\n-> "
        )
    )
    print(f"Is Array Sorted? -> {is_sorted(arr, 0, choice)}")


if __name__ == "__main__":
    main()
